Closure_Project/rest_api/serializers/CourseSerializer.py

```diff
@@ -1,11 +1,20 @@
+from django.db.models import Case, When
+
 from rest_api.serializers.DynamicSerializer import *
-from rest_api.models import Course
+from rest_api.models import Course, Student, CourseType
+from rest_api.utils import get_course_type
 
 
 class CourseSerializer(DynamicFieldsModelSerializer):
     pk = serializers.PrimaryKeyRelatedField(source='id', read_only=True)
+    type = serializers.SerializerMethodField(method_name='get_type')
+
+    def get_type(self, obj):
+        user = self.context.get('request').user
+        student = Student.objects.get(user=user)
+        return get_course_type(student.track, obj)
 
     class Meta:
         model = Course
-        fields = ('pk', 'course_id', 'data_year', 'name', 'semester', 'points', 'is_given_this_year',
+        fields = ('pk', 'course_id', 'data_year', 'name', 'semester', 'points', 'type', 'is_given_this_year',
```

Comment:
So now every course has its type, is there an `undefined` state, or is `Supplementary` just the de-facto default?

Reply:
When the track is undefined, the type field is null

Reply:
Is this `null` handled in other places that check for the course type?

Reply:
We don't use this type at all. it is simply for information to the client - when searching for a course, the user will know what it's type. Since all our users are registered students, they will see the type all the time and will not see a `null` value. 

We can use the API for administration with a superuser, and the type field will be null, and it will be OK.